### PR TITLE
Remove exception hiding in the importer

### PR DESCRIPTION
**Pull request by manuel-sommer**
This PR removes the code mentioned from @damiencarol in this issue https://github.com/DefectDojo/django-DefectDojo/issues/5714

Ref about the naming https://en.wikipedia.org/wiki/Error_hiding

### CODE REVIEW

**Comment by damiencarol** [comment]
@manuel-sommer the method `full_clean` does something with the object (modify values). Could you change the code to just call it and don't swallow exceptions that could pop up?

**Comment by manuel-sommer** [comment]
done, thank you for the explanation @damiencarol

**Comment by damiencarol** [comment]
The unit test code is wrong. The `lead` attribute is required. This was hidden by this small piece of code.

@manuel-sommer could you improve the unit test code by providing a value for `lead` attribute?

```
2022-01-12T11:11:10.6106908Z [36muwsgi_1         |[0m ======================================================================
2022-01-12T11:11:10.6107925Z [36muwsgi_1         |[0m FAIL: test_import_by_engagement_id (unittests.test_importers_importer.FlexibleImportTestAPI)
2022-01-12T11:11:10.6109201Z [36muwsgi_1         |[0m ----------------------------------------------------------------------
2022-01-12T11:11:10.6109939Z [36muwsgi_1         |[0m Traceback (most recent call last):
2022-01-12T11:11:10.6110882Z [36muwsgi_1         |[0m   File "/app/unittests/test_utils.py", line 126, in assertImportModelsCreated
2022-01-12T11:11:10.6111697Z [36muwsgi_1         |[0m     yield (
2022-01-12T11:11:10.6112724Z [36muwsgi_1         |[0m   File "/app/unittests/test_importers_importer.py", line 189, in test_import_by_engagement_id
2022-01-12T11:11:10.6114486Z [36muwsgi_1         |[0m     import0 = self.import_scan_with_params(NPM_AUDIT_NO_VULN_FILENAME, scan_type=NPM_AUDIT_SCAN_TYPE, engagement=self.engagement.id, test_title=TEST_TITLE_DEFAULT)
2022-01-12T11:11:10.6115799Z [36muwsgi_1         |[0m   File "/app/unittests/dojo_test_case.py", line 475, in import_scan_with_params
2022-01-12T11:11:10.6116897Z [36muwsgi_1         |[0m     return self.import_scan(payload, expected_http_status_code)
2022-01-12T11:11:10.6117954Z [36muwsgi_1         |[0m   File "/app/unittests/dojo_test_case.py", line 402, in import_scan
2022-01-12T11:11:10.6118998Z [36muwsgi_1         |[0m     self.assertEqual(expected_http_status_code, response.status_code, response.content[:1000])
2022-01-12T11:11:10.6120234Z [36muwsgi_1         |[0m   File "/usr/local/lib/python3.8/unittest/case.py", line 912, in assertEqual
2022-01-12T11:11:10.6121457Z [36muwsgi_1         |[0m     assertion_func(first, second, msg=msg)
2022-01-12T11:11:10.6122368Z [36muwsgi_1         |[0m AssertionError: 201 != 400 : b'{"message":"{\'lead\': [\'This field cannot be blank.\']}"}'
2022-01-12T11:11:10.6123446Z [36muwsgi_1         |[0m 
2022-01-12T11:11:10.6124728Z [36muwsgi_1         |[0m During handling of the above exception, another exception occurred:
2022-01-12T11:11:10.6125582Z [36muwsgi_1         |[0m 
2022-01-12T11:11:10.6126387Z [36muwsgi_1         |[0m Traceback (most recent call last):
2022-01-12T11:11:10.6127427Z [36muwsgi_1         |[0m   File "/usr/local/lib/python3.8/unittest/mock.py", line 1325, in patched
2022-01-12T11:11:10.6128294Z [36muwsgi_1         |[0m     return func(*newargs, **newkeywargs)
2022-01-12T11:11:10.6130011Z [36muwsgi_1         |[0m   File "/app/unittests/test_importers_importer.py", line 193, in test_import_by_engagement_id
2022-01-12T11:11:10.6131101Z [36muwsgi_1         |[0m     self.assertEqual(import0['product_id'], self.engagement.product.id)
2022-01-12T11:11:10.6132088Z [36muwsgi_1         |[0m   File "/usr/local/lib/python3.8/contextlib.py", line 131, in __exit__
2022-01-12T11:11:10.6132888Z [36muwsgi_1         |[0m     self.gen.throw(type, value, traceback)
2022-01-12T11:11:10.6133997Z [36muwsgi_1         |[0m   File "/app/unittests/test_utils.py", line 126, in assertImportModelsCreated
2022-01-12T11:11:10.6134777Z [36muwsgi_1         |[0m     yield (
2022-01-12T11:11:10.6135435Z [36muwsgi_1         |[0m   File "/app/unittests/test_utils.py", line 86, in __exit__
2022-01-12T11:11:10.6136178Z [36muwsgi_1         |[0m     self.test_case.assertEqual(
2022-01-12T11:11:10.6139152Z [36muwsgi_1         |[0m AssertionError: 0 != 1 : 0 <class 'dojo.models.Test'> objects created, 1 expected. query: SELECT `dojo_test`.`id`, `dojo_test`.`engagement_id`, `dojo_test`.`lead_id`, `dojo_test`.`test_type_id`, `dojo_test`.`scan_type`, `dojo_test`.`title`, `dojo_test`.`description`, `dojo_test`.`target_start`, `dojo_test`.`target_end`, `dojo_test`.`estimated_time`, `dojo_test`.`actual_time`, `dojo_test`.`percent_complete`, `dojo_test`.`environment_id`, `dojo_test`.`updated`, `dojo_test`.`created`, `dojo_test`.`version`, `dojo_test`.`build_id`, `dojo_test`.`commit_hash`, `dojo_test`.`branch_tag`, `dojo_test`.`api_scan_configuration_id` FROM `dojo_test`, first 100 objects: <TagulousCastTaggedQuerySet []>
```

**Comment by manuel-sommer** [comment]
Yeah, that's what I wanted to do @damiencarol. I already looked into it. I will just provide a string "admin".

**Comment by damiencarol** [comment]
@Maffooch I honestly prefer this version. I saw other one

**Comment by manuel-sommer** [comment]
@damiencarol Thank you for your contribution. I guess we are through here.

**Review by valentijnscholten — DISMISSED** [review]
can you pick a more neutral/relevant name for the migration file?

**Comment by damiencarol** [comment]
I agree with @valentijnscholten something like `remove_exception_swallowing` because it's exactly what we've done

**Comment by valentijnscholten** [comment]
`0144_lead_on_delete_set_null.py`

**Review by damiencarol — APPROVED** [review]

File: dojo/models.py [hunk 1]
```diff
@@ -1528,7 +1528,7 @@ class Meta:
 
 class Test(models.Model):
     engagement = models.ForeignKey(Engagement, editable=False, on_delete=models.CASCADE)
-    lead = models.ForeignKey(User, editable=True, null=True, on_delete=models.RESTRICT)
+    lead = models.ForeignKey(User, editable=True, null=True, blank=True, on_delete=models.SET_NULL)
```
Comment (StefanFl) [inline dojo/models.py:1531]: Why have you changed `on_delete=models.RESTRICT` with `on_delete=models.SET_NULL`? I don't think it's a good idea to change the lead in tests when deleting a user. When somebody wants to delete a user, they should look at all occurences of the user. That's why all ForeignKeys with `Dojo_User` or `User` are restricted.

**Review by StefanFl — CHANGES_REQUESTED** [review]

File: dojo/models.py (hunk 1 above)
Comment (damiencarol) [reply dojo/models.py:1531]: For now, the code of the importer/re-importer allow to have lead to be `None`. There is also no input for it in the UI, and in the API v2 you can push a report without a `lead` value. So for me it's better to consider it optional in the model to be aligned with the rest of the code and make it possible to remove the user without need to update every Test/engagement where the user was a lead. But I can change it. I don't think it's critical to this patch.

Comment (StefanFl) [reply dojo/models.py:1531]: It's fine for the `lead` to be optional. But please change it back to restricted, makes it consistent with the other occurences of users.

Comment (damiencarol) [reply dojo/models.py]: done

**Comment by damiencarol** [comment]
@valentijnscholten done for the neutral term. You can review it again

**Review by valentijnscholten — CHANGES_REQUESTED** [review]
please rename the migration

**Comment by damiencarol** [comment]
The migration is called `dojo/db_migrations/0146_unittest_remove_exception_swallowing.py` is it not good?

**Comment by StefanFl** [comment]
I think @valentijnscholten wants the name of the migration to describe what it changes, like `0146_lead_optional.py`. I think too this would be the better way.

**Review by valentijnscholten — APPROVED** [review]

**Review by StefanFl — APPROVED** [review]

**Pull request merged**